execute_orders.py

```diff
@@ -18,102 +19,87 @@
 
 
 def main(
-    symbol_name: str,
-    quantity: float,
-    price: str,
-    profit: float,
-    loss: float,
+    input_args: InputArgs
 ) -> None:
 
     client = Client(api_key=API_KEY, api_secret=SECRET_KEY)
-    symbol = client.get_symbol(symbol_name)
+    symbol = client.get_symbol(input_args.symbol)
 
     buy_order_type = "limit"
 
-    # To be used to check the Percent Filter rule
-    avg_price = client.get_avg_price(symbol)
-
-    # Get the precisions for both price and quantity
-    price_round = int(-np.log10(symbol.filters.price_filter.min_price))
-    # print("Min price decimal:", price_round)
-    qty_round = int(-np.log10(symbol.filters.lot_size_filter.min_qty))
-    # print("Min quantity decimal:", qty_round)
-
     # Place a market buy order
     buy_order, buy_quantity, buy_price = client.execute_buy_strategy(
         symbol,
         buy_order_type,
-        avg_price,
-        quantity,
-        price,
-        qty_round,
-        price_round
+        input_args.quantity,
+        input_args.price,
     )
 
     print(f"=> Buy price: {buy_price} {symbol.quote_asset}")
     print(
-        f"=> Total price: {round(float(buy_order['cummulativeQuoteQty']), price_round)} "
+        "=> Total price: "
+        f"{Decimal(buy_order['cummulativeQuoteQty'], symbol.price_decimal_precision)} "
```

Comment:
It sounds like Decimal is not taking a precision number as a 2nd argument but rather a Context.
What do you think of simply rounding `Decimal(buy_order['cummulativeQuoteQty'])`?

Reply:
It's fine if you want to keep the default precision found into buy_order.
Otherwise:
```suggestion
        f"{round(Decimal(buy_order['cummulativeQuoteQty']), symbol.price_decimal_precision)} "
```

Reply:
Your suggestion of using `round` sounds good, let's add it!